### ISSUE TITLE
Custom Unmarshal not working

### ISSUE THREAD

**Issue by Vingtoft** [amazon-ion/ion-go#188]
Following [this example](https://github.com/amazon-ion/ion-go/blob/master/ion/unmarshal_test.go#L319), I have written the below code:

```
package main

import (
    "bytes"
    "fmt"

    "github.com/amzn/ion-go/ion"
)

func main() {
    UnmarshalCustomMarshaler()
}

type unmarshalMe struct {
    Name   string
    custom bool
}

func (u *unmarshalMe) UnmarshalIon(r ion.Reader) error {
    fmt.Print("UnmarshalIon called")
    u.custom = true
    return nil
}

func UnmarshalCustomMarshaler() {
    ionBinary, err := ion.MarshalBinary(unmarshalMe{
        Name: "John Doe",
    })
    if err != nil {
        fmt.Println("Error marshalling ion binary: ", err)
        panic(err)
    }

    dec := ion.NewReader(bytes.NewReader(ionBinary))
    var decodedResult unmarshalMe

    ion.UnmarshalFrom(dec, &decodedResult)
    fmt.Println("Decoded result: ", decodedResult)
}
```

Expected result is that `u.custom` is true, this is not the case: `u.custom` is false.

Is this a bug, or have i misunderstanded  how to implement custom unmarshal?

**Comment by sjeandeaux** [comment]
Hi,

You should probably look at the version, you are using. 
The latest version v1.1.3 does not contain the feature.

**Comment by Vingtoft** [comment]
You are correct, thanks.

**Comment by popematt** [comment]
I've just released v1.2.0 which does contain the feature in #185.
